pkg/providers/vsphere/reconciler/reconciler.go

```
@@ -113,181 +96,47 @@ func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *an
-	specWithBundles *c.Spec, workloadTemplateNames, kubeadmconfigTemplateNames map[string]string,
-) (controller.Result, error) {
-	if !conditions.IsTrue(cluster, workerNodeSpecPlaneAppliedCondition) {
-		workersSpec, err := templateBuilder.GenerateCAPISpecWorkers(specWithBundles, workloadTemplateNames, kubeadmconfigTemplateNames)
-		if err != nil {
-			return controller.Result{}, err
-		}
-
-		if err := serverside.ReconcileYaml(ctx, r.client, workersSpec); err != nil {
-			return controller.Result{}, err
-		}
+func (r *Reconciler) ValidateMachineConfigs(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
+	_ = log.WithValues("phase", "validateMachineConfigs")
+	machineConfigs := clusterSpec.VSphereMachineConfigs
+	datacenterConfig := clusterSpec.VSphereDatacenter
 
-		conditions.MarkTrue(cluster, workerNodeSpecPlaneAppliedCondition)
-	}
-	return controller.Result{}, nil
-}
+	vsphereClusterSpec := vsphere.NewSpec(clusterSpec, machineConfigs, datacenterConfig)
 
-func (r *Reconciler) reconcileControlPlaneSpec(ctx context.Context, cluster *anywherev1.Cluster, templateBuilder providers.TemplateBuilder, specWithBundles *c.Spec, cpOpt func(values map[string]interface{}), log logr.Logger) (controller.Result, error) {
-	if !conditions.IsTrue(cluster, controlSpecPlaneAppliedCondition) {
-		log.Info("Applying control plane spec", "name", cluster.Name)
-		controlPlaneSpec, err := templateBuilder.GenerateCAPISpecControlPlane(specWithBundles, cpOpt)
-		if err != nil {
-			return controller.Result{}, err
-		}
-		if err := serverside.ReconcileYaml(ctx, r.client, controlPlaneSpec); err != nil {
-			return controller.Result{Result: &ctrl.Result{
-				RequeueAfter: defaultRequeueTime,
-			}}, err
-		}
-		conditions.MarkTrue(cluster, controlSpecPlaneAppliedCondition)
+	if err := r.validator.ValidateClusterMachineConfigs(ctx, vsphereClusterSpec); err != nil {
```

should we bubble up this error to the cluster status?